gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1222 @@
+
+    """
+    Update variational multinomial parameters, based on a document and a time-slice.
+    This is done based on the original Blei-LDA paper, where:
+    log_phi := beta * exp(Ψ(gamma)), over every topic for every word.
+    
+    TODO: incorporate lee-sueng trick used in **Lee, Seung: Algorithms for non-negative matrix factorization, NIPS 2001**.
+    """
+
+    K = lda_post.lda.num_topics
+    N = lda_post.doc.nterms
+
+    dig = numpy.zeros(K)
+
+    for k in range(0, K):
+        dig[k] = digamma(lda_post.gamma[k])
+
+    for n in range(0, N):
+        w = lda_post.doc.word[n]
+        for k in range(0, K):
+            lda_post.log_phi[n][k] = dig[k] + lda_post.lda.topics[w][k]
+
+        log_phi_row = lda_post.log_phi[n]
+        phi_row = lda_post.phi[n]
+
+        # log normalize
+        v = log_phi_row[0]
+        for i in range(1, len(log_phi_row)):
+            v = numpy.logaddexp(v, log_phi_row[i])
+
+        for i in range(0, len(log_phi_row)):
+            log_phi_row[i] = log_phi_row[i] - v
+
+        for k in range(0, K):
+            phi_row[k] = numpy.exp(log_phi_row[k])
+
+        lda_post.log_phi[n] = log_phi_row
+        lda_post.phi[n] = phi_row
+
+    return lda_post.phi, lda_post.log_phi
+
+def update_gamma(lda_post):
+
+    """
+    update variational dirichlet parameters as described in the original Blei LDA paper:
+    gamma = alpha + sum(phi), over every topic for every word.
+
+    """
+
+    K = lda_post.lda.num_topics
+    N = lda_post.doc.nterms
+
+    lda_post.gamma = numpy.copy(lda_post.lda.alpha)
+
+    for n in range(0, N):
+        phi_row = lda_post.phi[n]
+        count = lda_post.doc.count[n]
+
+        for k in range(0, K):
+            lda_post.gamma[k] += phi_row[k] * count
+
+    return lda_post.gamma
+
+def compute_expected_log_prob(sslm):
+
+    """
+    Compute the expected log probability given values of m.
+    The appendix describes the Expectation of log-probabilities in equation 5 of the DTM paper;
+    The below implementation is the result of solving the equation and is as implemented in the original Blei DTM code.
+    """
+
+    W = sslm.vocab_len
+    T = sslm.num_time_slices
+    for t in range(0, T):
+        for w in range(0, W):
+            sslm.e_log_prob[w][t] = sslm.mean[w][t + 1] - numpy.log(sslm.zeta[t])
+
+    return sslm.e_log_prob
+
+
+def sslm_counts_init(sslm, obs_variance, chain_variance, sstats):
+
+    """
+    Initialize State Space Language Model with LDA sufficient statistics.
+    """
+
+    W = sslm.vocab_len
+    T = sslm.num_time_slices
+
+    log_norm_counts = numpy.copy(sstats)
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+
+    log_norm_counts = log_norm_counts + 1.0 / W
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+    log_norm_counts = numpy.log(log_norm_counts)
+    
+
+    # setting variational observations to transformed counts
+    for t in range(0, T):
+        sslm.obs[:,t] = log_norm_counts
```

Comment:
Even if I use `numpy.vstack` I would have to iterate - because I am going column by column. Do you know any other way I can fill all columns of a matrix with the same values? @piskvorky any help here?